### PR TITLE
optimize/functions: implement the many local minima functions in the …

### PR DESCRIPTION
**Pull request by btracey**
…virtual library of simulation experiments

Updates #63

### CODE REVIEW

**Comment by btracey** [comment]
PTAL @vladimir-ch . They all follow pretty much the same format, so If you have changes to suggest in the first two (or so) I'll assume you mean global and fix up before you look at the rest.

File: optimize/functions/vlse.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+	for _, v := range x {
+		sum += v * math.Sin(math.Sqrt(math.Abs(v)))
+	}
+	return 418.9829*float64(len(x)) - sum
+}
+
+// Schaffer4 implements the fourth Shaffer function, a two dimensional function
+// with many local minima and many global minima. Its typical domain is the
+// hypercube of [-10, 10].
+//  f(x) = (sum_{i=1}^5 i cos((i+1)*x0 +i)) * (\sum_{i=1}^5 i cos((i+1)*x_1+i))
```
Comment (vladimir-ch) [inline optimize/functions/vlse.go]: nit: `x_0+i` (underscore and space)

File: optimize/functions/vlse.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package functions
+
+import "math"
+
+// This file implements functions from the Virtual Library of Simulation Experiments.
+//  https://www.sfu.ca/~ssurjano/bukin6.html
+// In many cases gradients and Hessians have been added. In some cases, these
```
Comment (vladimir-ch) [inline optimize/functions/vlse.go]: Not in this PR. Do you plan to add them later or for the remaining functions?

Comment (btracey) [inline optimize/functions/vlse.go]: Yea, I have gradients for Rastrigen that I at least plan to add.

**Pull request merged**